CMU-313/nodebb-f24-bluesleep
# Anonymous Feature (Sprint #2) - implementing anonymous field in the backend

### Backend Feature: Implementing the Anonymous Field 

**Summary**
This PR tackles Issue #12, and we attempted to implement a field handling the anonymous nature of a post in the backend, while being connected to the frontend. 

---
### Key Changes: 

**1) Adding anonymous field to frontend post handler function**
- nodebb-plugin-composer-default/static/lib/composer.js) 

**2) Adding anonymous field to the backend:**
- src/posts/create.js
- src/posts/data.js
- src/posts/summary.js
- src/topics/create.js

**3) Adding anonymous field to the schema:**
- public/openapi/components/schemas/PostObject.yaml
- public/openapi/components/schemas/TopicObject.yaml
- public/openapi/read/topic/topic_id.yaml
- public/openapi/write/posts/pid.yaml 

--- 
### Challenges Encountered: 
1. Errors matching up schema with response body 
![Screenshot 2024-10-10 at 10 45 09 PM](https://github.com/user-attachments/assets/86026596-2142-436f-a282-bea1c56685bd)
![Screenshot 2024-10-10 at 10 45 54 PM](https://github.com/user-attachments/assets/e7527488-9708-48d4-af3e-06403ca652b1)

--- 
### Current Status: 
**Build status:** The manual lint fixes are in place, and the code should now comply with the project’s coding standards.
**Feature status:** 
- The checkbox is present and functional on the frontend 
- Running into test suite errors on schema matching response body (our implementation of the anonymous field in the backend) 
**Next Steps:** Further examination of what functions we implemented our anonymous field triggered the schema errors. 

Current status: The anonymous field is included in the post function handling the frontend UI for drafting posts. This field is able to be connected in the backend. Running into issues with having the backend match with the schema right now.

--- 
### Testing Instructions: 
1. Checkout the `anonymous-feature` branch.
2. Run npm install and ensure the project builds successfully.
3. Test the anonymous checkbox functionality by toggling it within a draft post
4. Verify that the code adheres to linting standards
